sherpa/astro/io/pyfits_backend.py

```diff
@@ -686,7 +682,7 @@ def get_rmf_data(arg, make_copy=False):
             data['matrix'] = hdu.data.field('MATRIX')
 
         data['header'] = _get_meta_data(hdu)
-        data['header'].pop('DETCHANS')
+        data['header'].pop('DETCHANS', None)
```

Comment:
Obviously some of these "change to pop(foo, None)" are good as it simplifies the code (removes the try/except handing) but in some cases like this I think the file is invalid if it doesn't contain the keyword, so we technically don't need the safety check.

In this case a few lines earlier we have

        data['detchans'] = SherpaUInt(_require_key(hdu, 'DETCHANS'))

and I would hope there's an equivalent in the crates version but can not guarantee that as that could be deep in the crates code which we don't get to see.

I'm not requesting a change here, just bringnig it up for a small discussion.

Reply:
I usually try to flexible when reading and strict when writing. For Sherpa, there is no harm done if, for whatever reason, header has not DETCHANS keyword at this point, since we are not using the popped value anyway, so we might as well accept a situation where it's not there. Plus, this is just replicating the old behavior as the old Meta always returned `None` if a keyword was not found. 

In some instances (I'm not sure know if this was one of them) there are code path which pop a keyword earlier and tests actually fail with a "KeyError" in one of the those pop commands, where it's a totally unnecessary fail because the return vaue of the pop is not used. Refactoring those code path is beyond the scope of this PR.